vaadin-context-menu-flow/src/main/resources/META-INF/resources/frontend/contextMenuConnector.js

```diff
@@ -1,108 +1,114 @@
+  const tryCatchWrapper = function (callback) {
+    return window.Vaadin.Flow.tryCatchWrapper(callback, 'Vaadin Context Menu', 'vaadin-context-menu-flow');
+  };
+
+  // Not using ES6 imports in this file yet because the connector in V14 must
+  // still work in Legacy bower projects. See: `contextMenuConnector-es6.js` for
+  // the Polymer3 approach.
+  window.Vaadin.Flow.Legacy = window.Vaadin.Flow.Legacy || {};
+
+  window.Vaadin.Flow.contextMenuConnector = {
+
+    // NOTE: This is for the TARGET component, not for the <vaadin-context-menu> itself
+    init: tryCatchWrapper(function(target) {
+      if (target.$contextMenuConnector) {
+        return;
+      }
 
-    if (window.Polymer) {
+      if (window.Polymer) {
         // Polymer2 approach.
         window.Vaadin.Flow.Legacy.GestureEventListeners = window.Vaadin.Flow.Legacy.GestureEventListeners || Polymer.GestureEventListeners;
         window.Vaadin.Flow.Legacy.Gestures = window.Vaadin.Flow.Legacy.Gestures || Polymer.Gestures;
-    } else if (!window.Vaadin.Flow.Legacy.Gestures) {
-      console.log("ContextMenu is unable to load Polymer helpers.");
-      return;
-    }
-
-    const GestureEventListeners = window.Vaadin.Flow.Legacy.GestureEventListeners;
-    const Gestures = window.Vaadin.Flow.Legacy.Gestures;
-
-    target.$contextMenuConnector = {
-
-      openOnHandler: function(e) {
-        e.preventDefault();
-        e.stopPropagation();
-        this.$contextMenuConnector.openEvent = e;
-        let detail = {};
-        if (target.getContextMenuBeforeOpenDetail) {
-          detail = target.getContextMenuBeforeOpenDetail(e);
-        }
-        target.dispatchEvent(new CustomEvent('vaadin-context-menu-before-open', {detail: detail}));
-      },
+      } else if (!window.Vaadin.Flow.Legacy.Gestures) {
+        console.log("ContextMenu is unable to load Polymer helpers.");
+        return;
+      }
 
-      updateOpenOn: function(eventType) {
-        this.removeListener();
-        this.openOnEventType = eventType;
+      const GestureEventListeners = window.Vaadin.Flow.Legacy.GestureEventListeners;
+      const Gestures = window.Vaadin.Flow.Legacy.Gestures;
 
-        customElements.whenDefined('vaadin-context-menu').then(() => {
-          if (Gestures.gestures[eventType]) {
+      target.$contextMenuConnector = {
+
+        openOnHandler: tryCatchWrapper(function(e) {
+          e.preventDefault();
+          e.stopPropagation();
+          this.$contextMenuConnector.openEvent = e;
+          let detail = {};
+          if (target.getContextMenuBeforeOpenDetail) {
+            detail = target.getContextMenuBeforeOpenDetail(e);
+          }
+          target.dispatchEvent(new CustomEvent('vaadin-context-menu-before-open', {detail: detail}));
+        }),
+
+        updateOpenOn: tryCatchWrapper(function(eventType) {
+          this.removeListener();
+          this.openOnEventType = eventType;
+
+          customElements.whenDefined('vaadin-context-menu').then(() => {
```

Comment:
One more thing. Should this arrow function also be wrapped with `tryCatchWrapper` as it's triggered from `customElements.whenDefined()`?

Reply:
Updated.